vortex-fastlanes/src/bitpacking/compute.rs

```diff
@@ -24,6 +31,57 @@ impl FlattenFn for BitPackedArray {
 
 impl TakeFn for BitPackedArray {
     fn take(&self, indices: &dyn Array) -> VortexResult<ArrayRef> {
-        take(&flatten(self)?.into_array(), indices)
+        let prim_indices = flatten_primitive(indices)?;
+        // Group indices into 1024 chunks and relativise them to the beginning of each chunk
+        let relative_indices: Vec<(usize, Vec<u16>)> = match_each_integer_ptype!(prim_indices.ptype(), |$P| {
+            let groupped_indices = prim_indices
+                .typed_data::<$P>()
+                .iter()
+                .group_by(|idx| (**idx / 1024) as usize);
+            groupped_indices
+                .into_iter()
+                .map(|(k, g)| (k, g.map(|idx| (*idx % 1024) as u16).collect()))
+                .collect()
+        });
+
+        let taken = relative_indices
```

Comment:
I think we need to do something slightly different here. Currently, the grouped indices are only grouped into consecutive runs. That means if I have indices `[0, 2000, 0, 2000, 0, 2000]` I will call bitunpack on two chunks a total of 6 times. 

Since we know the result size, we could allocate the result vector, then unpack each chunk once. For each {chunk, relative_idx} pair we store a vec of result_idx so we can iterate chunks and write directly into the result.

Reply:
I think we just need a different group by